test/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-project(tests VERSION 0.1.0)
+enable_testing()

```

Comment:
I guess if this is working now maybe no reason to change it but `Catch2` does do automatic `ctest` registration. I have played around with it in a bit but here is the link: https://github.com/catchorg/Catch2/blob/devel/docs/cmake-integration.md#automatic-test-registration. If you think it is worth it let me know. I can also take a stab at it after we merge this.

Reply:
This would change the below script where you do the `add_test`

Reply:
Ah I did not know this, thanks. I think you can also directly commit to my branch if I am not wrong?

Reply:
Yea I can check out your and push to it. Wasn't sure if you want changes on it or PR to merge. Didn't want to interrupt the workflow. I can make this change if that is easier.

Reply:
fixed in b48f4730f62f3e4c2ba3c53f127e23fa5748c08b